### PR TITLE
Allow check if ArgumentParser has parsed values

### PR DESCRIPTION
**Pull request by skrobinson**

### CODE REVIEW

File: include/argparse/argparse.hpp [hunk 1]
```diff
@@ -1104,6 +1104,21 @@ class ArgumentParser {
...
+                                [](auto &it) {
+                                    return it.second->m_is_used;
+                                });
+    auto subparser_used = std::any_of(m_subparser_used.cbegin(),
+                                      m_subparser_used.cend(),
+                                      [](auto &it) {
+                                          return it.second;
+                                      });
+
+    return m_is_parsed && (arg_used || subparser_used);
```
Comment (SergiusTheBest) [inline include/argparse/argparse.hpp:1119]: Why don't we use just `m_is_parsed`? I have a subcommand without arguments and `operator bool()` returns false for it.

**Comment by skrobinson** [comment]
@SergiusTheBest The question I'm trying to answer with this `operator bool()` is: has this parser (or it's subparsers) run **and** found values. Using only `m_is_parsed` would only indicate the parser ran without an error, not that any values were parsed.

For the case where subparsers are commands and don't require any arguments the `ArgumentParser::is_subcommand_used()` function may better match your needs.

**Comment by SergiusTheBest** [comment]
@skrobinson I understand. But it's not so obvious as `operator bool()` is very generic and is better suited for `m_is_parsed` than  `m_is_parsed && (arg_used || subparser_used)` (taking into account that arguments are optional and not required for a command).

**Comment by SergiusTheBest** [comment]
@skrobinson BTW, is a test case required for contributing? I'd like to add `ArgumentParser::is_subcommand_used(const ArgumentParser& subcommand)`.

**Comment by skrobinson** [comment]
Test cases are encouraged. You could probably add to and adapt tests in `test/test_subparsers.cpp` rather than writing new tests.

I believe your new overload is a good idea.

**Comment by SergiusTheBest** [comment]
Ok, thanks for the hint.

**Pull request merged**